---
title: "Lab_10"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

```{r}
library(vcfR)
```

```{r}
data(vcfR_example)
vcf
```
```{r}
strwrap(vcf@meta[1:7])

```

```{r}
queryMETA(vcf)
```

```{r}
queryMETA(vcf, element = 'DP')
```

```{r}
queryMETA(vcf, element = 'FORMAT=<ID=DP')
```

```{r}
head(getFIX(vcf))
```

```{r}
vcf@gt[1:6, 1:4]
```

```{r}
vcf <- read.vcfR("data/pinfsc50_filtered.vcf.gz")
```

```{r}
head(vcf)
```

```{r}
write.vcf(vcf, "myVCFdata_filtered.vcf.gz")
```

### Exercises part 1

### 1)  How would we find more information about read.vcfR()? 

### The line of code below will pull up the help tab for more information on read.vcfR.

```{r}
?read.vcfR
```

### 2) How would we learn what the acronym “AD” stands for?

```{r}
queryMETA(vcf, element = 'AD')


```

### 3) We used the head() function to view the first few lines of fix data. How would we view the last few lines of fix data?

```{r}
tail(vcf@fix)
```

### 4) There is a column in the fix portion of the data called QUAL. It is not defined in the meta portion of the data because it is defined in the VCF specification. It stands for ‘quality’. Does QUAL appear useful to us? Why or why not? Yes, as the histogram shows.

```{r}
plot(vcf)
```
```{r}
# Alternate ggplot2 answer.
library(ggplot2)
qplot(getQUAL(vcf), geom = "histogram")
```

### 5) How would we query the sample names?

```{r}
colnames(vcf@gt)
```

### Part II - Analysis of Genome Data

```{r}
library('vcfR')
vcf <- read.vcfR("data/pinfsc50_filtered.vcf.gz")
# This data can also be download from the course repo - https://github.com/jeffreyblanchard/EvoGeno/blob/master/Grunwald/pinfsc50_filtered.vcf.gz
```

```{r}
vcf
```



```{r}
x <- vcfR2genlight(vcf)
```

```{r}
x
```

```{r}
# vcfR
gt <- extract.gt(vcf, element = "GT")
gt[c(2,6,18), 1:3]
```

```{r}
# genlight
t(as.matrix(x))[c(1,5,17), 1:3]
```

```{r}
library(adegenet)
```

```{r}
pop(x) <- as.factor(c("us", "eu", "us", "af", "eu", "us", "mx", "eu", "eu", "sa", "mx", "sa", "us", "sa", "Pmir", "us", "eu", "eu"))
popNames(x)
```

```{r}
ploidy(x) <- 2
```

```{r}
x.dist <- dist(x)
```



```{r}
x.dist <- poppr::bitwise.dist(x)
```


```{r}
library(vcfR)

# Find the files.
vcf_file <- system.file("extdata", "pinf_sc50.vcf.gz", package = "pinfsc50")
dna_file <- system.file("extdata", "pinf_sc50.fasta", package = "pinfsc50")
gff_file <- system.file("extdata", "pinf_sc50.gff", package = "pinfsc50")

# Input the files.
vcf <- read.vcfR(vcf_file, verbose = FALSE)
dna <- ape::read.dna(dna_file, format = "fasta")
gff <- read.table(gff_file, sep="\t", quote="")

# Create a chromR object.
chrom <- create.chromR(name="Supercontig", vcf=vcf, seq=dna, ann=gff, verbose=TRUE)
```

```{r}
chrom
```

```{r}
plot(chrom)
```

```{r}
chromoqc(chrom, dp.alpha = 66)
```
```{r}
chrom <- proc.chromR(chrom, verbose = TRUE)
```

```{r}
plot(chrom)
```

```{r}
chromoqc(chrom, dp.alpha = 66)
```

```{r}
#vcf <- read.vcfR("pinfsc50_qc.vcf.gz", verbose = FALSE)
vcf <- read.vcfR("data/pinfsc50_filtered.vcf.gz", verbose = FALSE)
chrom <- create.chromR(name="Supercontig", vcf=vcf, seq=dna, ann=gff, verbose=FALSE)
chrom <- proc.chromR(chrom, verbose = FALSE)
chromoqc(chrom, dp.alpha = 66)
```
```{r}
head(chrom@var.info)
```
```{r}
head(chrom@win.info)
```
```{r}
library(vcfR)
data(vcfR_example)
pop <- as.factor(c("us", "eu", "us", "af", "eu", "us", "mx", "eu", "eu", "sa", "mx", "sa", "us", "sa", "Pmir", "us", "eu", "eu"))
myDiff <- genetic_diff(vcf, pops = pop, method = 'nei')
knitr::kable(head(myDiff[,1:15]))
```

```{r}
knitr::kable(head(myDiff[,16:19]))
```

```{r}
knitr::kable(round(colMeans(myDiff[,c(3:9,16,19)], na.rm = TRUE), digits = 3))
```

```{r}
library(reshape2)
library(ggplot2)

dpf <- melt(myDiff[,c(3:8,19)], varnames=c('Index', 'Sample'), value.name = 'Depth', na.rm=TRUE)

```

```{r}
p <- ggplot(dpf, aes(x=variable, y=Depth)) + geom_violin(fill="#2ca25f", adjust = 1.2)
p <- p + xlab("")
p <- p + ylab("")
p <- p + theme_bw()
p
```
### Exercises Part II

### 1) You actually have everything you need to make a Manhattan plot. Can you figure out how to plot G′ST (y-axis) by genomic position (POS)?


```{r}
plot(getPOS(vcf), myDiff$Gprimest,  pch = 20, col = "#1E90FF44", xlab = "", ylab = "", ylim = c(0, 1), xaxt = "n")
axis(side = 1, at = seq(0, 1e5, by = 1e4), labels = seq(0, 100, by = 10))
title(xlab='Genomic position (Kbp)')
title(ylab = expression(italic("G'"["ST"])))
```
### 2) This Manhatttan plot should look a bit unusual. Can you think of anything that may be wrong with this analysis? The table looks unusual due to the very small sample size.

```{r}
table(pop)
```

### 3) Can you figure out how to zoom in on a particular region of a chromosome in chromoqc()?

```{r}
chromoqc(chrom, dp.alpha = 66, xlim = c(2e05, 4e05))
```

### 4) Can you use the function queryMETA() to look for other data in your file that may be of interest?

```{r}
queryMETA(vcf)
```

### References 

### Danecek, Petr, Adam Auton, Goncalo Abecasis, Cornelis A Albers, Eric Banks, Mark A DePristo, Robert E Handsaker, et al. 2011. “The Variant Call Format and VCFtools.” Bioinformatics 27 (15): 2156–8. https://doi.org/10.1093/bioinformatics/btr330.

### Hedrick, Philip W. 2005. “A Standardized Genetic Differentiation Measure.” Evolution 59 (8): 1633–8. http://dx.doi.org/10.1111/j.0014-3820.2005.tb01814.x.

### Jombart, Thibaut. 2008. “adegenet: A R Package for the Multivariate Analysis of Genetic Markers.” Bioinformatics 24 (11): 1403–5. https://doi.org/10.1093/bioinformatics/btn129.

### Jost, Lou. 2008. “GST And Its Relatives Do Not Measure Differentiation.” Molecular Ecology 17 (18): 4015–26. http://dx.doi.org/10.1111/j.1365-294X.2008.03887.x.

### Kamvar, Zhian N, Jonah C Brooks, and Niklaus J Grünwald. 2015. “Novel R tools for analysis of genome-wide population genetic data with emphasis on clonality.” Name: Frontiers in Genetics 6: 208. https://doi.org/10.3389/fgene.2015.00208.

### Kamvar, Z N, J F Tabima, and Niklaus J Grünwald. 2014. “Poppr: An R Package for Genetic Analysis of Populations with Clonal, Partially Clonal, and/or Sexual Reproduction.” PeerJ 2: e281. https://doi.org/10.7717/peerj.281.

### Knaus, Brian J, and Niklaus J Grünwald. 2017. “Vcfr: A Package to Manipulate and Visualize Variant Call Format Data in R.” Molecular Ecology Resources 17 (1): 44–53. http://dx.doi.org/10.1111/1755-0998.12549.

### Nei, Masatoshi. 1973. “Analysis of Gene Diversity in Subdivided Populations.” Proceedings of the National Academy of Sciences 70 (12): 3321–3. http://www.pnas.org/content/70/12/3321.abstract.

### Simpson, Edward H. 1949. “Measurement of Diversity.” Nature 163: 688. http://dx.doi.org/10.1038/163688a0.

### Wright, Sewall. 1949. “The Genetical Structure of Populations.” Annals of Eugenics 15 (1): 323–54. https://doi.org/10.1111/j.1469-1809.1949.tb02451.x.