---
title: "Lab_8"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

### Exercise 1

### Through the use of NGPhylogeny.fr I was able to create this phylogenic tree that displays the DNA sequences that I took from our Moodle site.

```{r}
library(knitr)
```

```{r}
knitr::include_graphics('Image tree Genomics png.png')
```


```{r}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install(version = "3.11")
```

```{r}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("ggtree")
```

```{r}
library(tidyverse)
```

```{r}
library(ggtree)

```

```{r}
browseVignettes("ggtree")
```



```{r}
library(ape)
```


```{r}
library(tidyverse)
library(ggtree)

tree <- read.tree("data/fastree_result (1).tre(final)")
tree
```

```{r}
# build a ggplot with a geom_tree
ggplot(tree) + geom_tree() + theme_tree()

# This is convenient shorthand
ggtree(tree)

```

```{r}
# add a scale
ggtree(tree) + geom_treescale()
# or add the entire scale to the x axis with theme_tree2()
ggtree(tree) + theme_tree2()

```

```{r}
ggtree(tree, branch.length="none")
```

```{r}
ggtree(tree, branch.length="none", color="blue", size=2, linetype=3)
```
### Exercise 1

```{r}
ggtree(tree, layout = "slanted")
```


```{r}
ggtree(tree, layout = "circular")
```

```{r}
ggtree(tree, branch.tree ='none', color = "red", size=2, linetype =19)
```


```{r}
# create the basic plot
p <- ggtree(tree)
# add node points
p + geom_nodepoint()
# add tip points
p + geom_tippoint()
# Label the tips
p + geom_tiplab()
```

### Exercise 2

```{r}
p <- ggtree(tree)
p + ggtitle("Exercise 2 phylogent changing nodes and tips")
p + geom_nodepoint()
p + geom_tippoint()
p + geom_tiplab()
p + geom_point(aes(subset=(node==all())), shape=23, size=5, fill="purple") +
  geom_nodepoint(shape=21, fill="yellow")
 
```



```{r}
ggtree(tree) + geom_text(aes(label=node), hjust=-.3)
```
```{r}
ggtree(tree) + geom_tiplab()
```


```{r}
ggtree(tree) + 
  geom_cladelabel(node=17, label="Some random clade", color="red")
```
```{r}
ggtree(tree) + 
  geom_tiplab() + 
  geom_cladelabel(node=17, label="Some random clade", 
                  color="red2", offset=.8)
```

```{r}
ggtree(tree) + 
  geom_tiplab() + 
  geom_cladelabel(node=17, label="Some random clade", 
                  color="red2", offset=.8) + 
  geom_cladelabel(node=21, label="A different clade", 
                  color="blue", offset=.8)

```
```{r}
ggtree(tree) + 
  geom_tiplab() + 
  geom_cladelabel(node=17, label="Some random clade", 
                  color="red2", offset=.8, align=TRUE) + 
  geom_cladelabel(node=21, label="A different clade", 
                  color="blue", offset=.8, align=TRUE) + 
  theme_tree2() + 
  xlim(0, 70) + 
  theme_tree()

```
```{r}
ggtree(tree) + 
  geom_tiplab() + 
  geom_hilight(node=30, fill="gold", extend = 0.5) + 
  geom_hilight(node=37, fill="purple", extend = 0.5)
```


```{r}
ggtree(tree) + 
  geom_tiplab() + 
  geom_taxalink("E", "H", color="blue3") +
  geom_taxalink("C", "G", color="orange2", curvature=-.9)
```
```{r}
library(treeio)
library(ggimage)
```



```{r}
# Read the data
tree <- read.beast("data/flu_tree_beast.tree")
# supply a most recent sampling date so you get the dates
# and add a scale bar
ggtree(tree, mrsd="2013-01-01") + 
  theme_tree2() 
# Finally, add tip labels and adjust axis
ggtree(tree, mrsd="2013-01-01") + 
  theme_tree2() + 
  geom_tiplab(align=TRUE, linesize=.5) + 
  xlim(1990, 2020)
```

```{r}
msaplot(p=ggtree(tree), fasta="data/flu_aasequence.fasta", window=c(150, 175))
```


```{r}
set.seed(42)
trees <- lapply(rep(c(10, 25, 50, 100), 3), rtree)
class(trees) <- "multiPhylo"
ggtree(trees) + facet_wrap(~.id, scale="free", ncol=4) + ggtitle("Many trees. Such phylogenetics. Wow.")
```

```{r}
# Generate a random tree with 30 tips
tree <- rtree(30)
# Make the original plot
p <- ggtree(tree)
# generate some random values for each tip label in the data
d1 <- data.frame(id=tree$tip.label, val=rnorm(30, sd=3))
# Make a second plot with the original, naming the new plot "dot", 
# using the data you just created, with a point geom.
p2 <- facet_plot(p, panel="dot", data=d1, geom=geom_point, aes(x=val), color='red3')
# Make some more data with another random value.
d2 <- data.frame(id=tree$tip.label, value = abs(rnorm(30, mean=100, sd=50)))
# Now add to that second plot, this time using the new d2 data above, 
# This time showing a bar segment, size 3, colored blue.
p3 <- facet_plot(p2, panel='bar', data=d2, geom=geom_segment, 
           aes(x=0, xend=value, y=y, yend=y), size=3, color='blue4') 
# Show all three plots with a scale
p3 + theme_tree2()
```

```{r}
library(magick)
```

```{r}
# get phylopic 

newick <- "((Pongo_abelii,(Gorilla_gorilla_gorilla,(Pan_paniscus,Pan_troglodytes)Pan,Homo_sapiens)Homininae)Hominidae,Nomascus_leucogenys)Hominoidea;"

tree <- read.tree(text=newick)

d <- ggimage::phylopic_uid(tree$tip.label)
d$body_mass = c(52, 114, 47, 45, 58, 6)

p <- ggtree(tree) %<+% d + 
  geom_tiplab(aes(image=uid, colour=body_mass), geom="phylopic", offset=2.5) +
  geom_tiplab(aes(label=label), offset = .2) + xlim(NA, 7) +
  scale_color_viridis_c()
p  
```

### Exercise 4

```{r}
library(tidyverse)
library(ggtree)

tree <- read.tree("data/fastree_result (1).tre(final)")
tree
```

```{r}
p <- ggtree(tree, branch.length="none", color="red", size=2, linetype=3)

p + geom_nodepoint()

p + geom_tippoint()

p + geom_tiplab()
 ggtree(tree) +
  geom_nodelab()  +
 geom_tiplab(hjust=-.1, size = 3) +
  geom_hilight(node=30, fill="gold", extend = 0.5)  +
  geom_hilight(node=37, fill="purple", extend = 0.5) +
geom_hilight(node=25, fill="blue", extend = 0.5) +
geom_hilight(node=16, fill="purple", extend = 0.75) +
  geom_hilight(node=1, fill="purple", extend = 0.5)
```

